packages/datadog-instrumentations/src/pino.js

```diff
@@ -0,0 +1,102 @@
+'use strict'
+
+const {
+  channel,
+  addHook
+} = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+function wrapPino (symbol, wrapper, pino) {
+  return function pinoWithTrace () {
+    const instance = pino.apply(this, arguments)
+
+    Object.defineProperty(instance, symbol, {
+      configurable: true,
+      enumerable: true,
+      writable: true,
+      value: wrapper(instance[symbol])
+    })
+
+    return instance
+  }
+}
+
+function wrapAsJson (asJson) {
+  const ch = channel('apm:pino:log')
+  return function asJsonWithTrace (obj, msg, num, time) {
+    obj = arguments[0] = obj || {}
+
+    const payload = { logMessage: obj, receiver: null }
```

Comment:
Receiver takes in a proxy object, through the proxy mechanism first used in the winston plugin. We don't always need a proxy (as per previous code that doesn't need it (although we should revisit some of that)) and so this is an attempt to not create a proxy when we don't have to.

Reply:
That is, since they're all using the same subscriber (effectively), we need some way of notifying it that a proxy is necessary, so I'm doing that by adding the receiver property.